src/types/num.rs

```diff
@@ -20,6 +20,21 @@ pyobject_native_type_core!(PyInt, pyobject_native_static_type_object!(ffi::PyLon
 #[deprecated(since = "0.23.0", note = "use `PyInt` instead")]
 pub type PyLong = PyInt;
 
+impl PyInt {
+    /// Creates a new Python int object.
+    ///
+    /// Panics if out of memory.
+    pub fn new<
+        'a,
+        T: IntoPyObject<'a, Target = PyInt, Output = Bound<'a, PyInt>, Error = Infallible>,
+    >(
+        py: Python<'a>,
+        i: T,
+    ) -> Bound<'a, PyInt> {
+        T::into_pyobject(i, py).unwrap()
```

Comment:
```suggestion
        match T::into_pyobject(i, py) {
            Ok(v) => v,
            Err(never) => match never {},
        }
```

Probably not too important here, but we can statically ensure there will be no panic here.